src/cmd/build.js

```diff
@@ -191,21 +191,47 @@ export default async function build(
 
 export type FileFilterOptions = {
   filesToIgnore?: Array<string>,
+  sourceDir?: string
 };
 
 /*
  * Allows or ignores files when creating a ZIP archive.
  */
 export class FileFilter {
-  filesToIgnore: Array<string>;
-
-  constructor({filesToIgnore}: FileFilterOptions = {}) {
-    this.filesToIgnore = filesToIgnore || [
-      '**/*.xpi',
-      '**/*.zip',
-      '**/.*', // any hidden file
-      '**/node_modules',
-    ];
+  _ignore: Object;
+
+  constructor({filesToIgnore, sourceDir}: FileFilterOptions = {}) {
+    this._ignore = createIgnore();
+
+    // TODO if a xpiignore file is provided, then
+    // are defaults required?
+    this._ignore.add(filesToIgnore || [
+      '*.xpi',
+      '*.zip',
+      '.*', // any hidden file
+      'node_modules',
+    ]);
+
+    if (sourceDir) {
+      const ignoreFilePath = path.join(sourceDir, '.xpiignore');
```

Comment:
I had little free time last month so I couldn't follow up on this PR. It seems the project is opting for a `*.js` file and not using an ignore file I believe.